L1Trigger/L1TMuonEndCap/plugins/L1TMuonEndCapShowerProducer.cc

```diff
@@ -23,16 +39,9 @@ void L1TMuonEndCapShowerProducer::produce(edm::Event& iEvent, const edm::EventSe
 
   // ___________________________________________________________________________
   // Run the sector processors
-
-  for (int endcap = emtf::MIN_ENDCAP; endcap <= emtf::MAX_ENDCAP; ++endcap) {
-    for (int sector = emtf::MIN_TRIGSECTOR; sector <= emtf::MAX_TRIGSECTOR; ++sector) {
-      const int es = (endcap - emtf::MIN_ENDCAP) * (emtf::MAX_TRIGSECTOR - emtf::MIN_TRIGSECTOR + 1) +
-                     (sector - emtf::MIN_TRIGSECTOR);
-
-      sector_processors_shower_.at(es).configure(config_, endcap, sector);
-      sector_processors_shower_.at(es).process(showers, *out_showers);
-    }
-  }
+  forEachProcessor([&](const int endcap, const int sector, const int es) {
+    sector_processors_shower_.at(es).process(showers, *out_showers);
```

Comment:
I'm also wondering why there needs to be more than one `SectorProcessShower`. This function could take the `endcap` and `sector` integers as arguments, and the configuration parameters taken from the `ParameterSet` are the same for all objects. But I leave this to the domain experts.

Reply:
Hi @makortel , this is written in the same way as the actual EMTF sector processors. It's technically possible to have different configurations on each sector processor, in case we need to recover for some missing/faulty chambers in that sector's coverage. For now, all the sectors are using the same config so it's not used. However, I would still keep it in place if for some reason something changes in the future.